services/swagger/swagger.service.js

```diff
@@ -1,67 +1,69 @@
+const validatorSchema = Joi.object()
+  .keys({
+    schemaValidationMessages: Joi.array().items(
+      Joi.object({
+        level: Joi.string().required(),
+        message: Joi.string().required(),
+      }).required()
+    ),
+  })
+  .required()
 
-        const badgeData = getBadgeData('swagger', data)
+module.exports = class SwaggerValidatorService extends BaseJsonService {
+  static render({ message, clr }) {
+    return { message: message, color: clr }
+  }
+
+  static get url() {
+    return {
+      base: 'swagger/valid/2.0',
+      format: '(http(?:s)?)/(.+)',
+      capture: ['protocol', 'url'],
+    }
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'swagger' }
+  }
 
-        const urlParam = encodeURIComponent(scheme + '://' + swaggerUrl)
-        const url = 'http://online.swagger.io/validator/debug?url=' + urlParam
-        const options = {
-          method: 'GET',
-          url: url,
-          gzip: true,
-          json: true,
-        }
-        request(options, (err, res, json) => {
-          try {
-            if (
-              err != null ||
-              res.statusCode >= 500 ||
-              typeof json !== 'object'
-            ) {
-              badgeData.text[1] = 'inaccessible'
-              sendBadge(format, badgeData)
-              return
-            }
+  async handle({ protocol, url }) {
+    const json = await this.fetch({ protocol, urlF: url })
+    const valMessages = json.schemaValidationMessages
 
-            const messages = json.schemaValidationMessages
-            if (messages == null || messages.length === 0) {
-              badgeData.colorscheme = 'brightgreen'
-              badgeData.text[1] = 'valid'
-            } else {
-              badgeData.colorscheme = 'red'
+    if (!valMessages || valMessages.length === 0) {
+      return this.constructor.render({ message: 'valid', clr: 'brightgreen' })
+    } else {
+      return this.constructor.render({ message: 'invalid', clr: 'red' })
+    }
+  }
+
+  async fetch({ protocol, urlF }) {
+    const url = `http://online.swagger.io/validator/debug?url=${protocol}://${urlF}`
+    return this._requestJson({
+      url,
+      schema: validatorSchema,
+    })
+  }
+
+  static get category() {
+    return 'other'
+  }
 
-              const firstMessage = messages[0]
-              if (
-                messages.length === 1 &&
-                firstMessage.level === 'error' &&
-                /^Can't read from/.test(firstMessage.message)
-              ) {
-                badgeData.text[1] = 'not found'
-              } else {
-                badgeData.text[1] = 'invalid'
-              }
-            }
-            sendBadge(format, badgeData)
-          } catch (e) {
-            badgeData.text[1] = 'inaccessible'
-            sendBadge(format, badgeData)
-          }
-        })
-      })
-    )
+  static get examples() {
+    return [
+      {
+        title: 'Swagger Validator',
+        urlPattern: 'https/:url',
```

Comment:
In our Jenkins coverage integration, we refer to http/https as "scheme". Would it be a good idea to make things consistent?